frontend/src/components/layout/Header.tsx

```diff
@@ -9,80 +9,88 @@
+import { Box, Text, ColorModeSwitch, Flex, Breadcrumbs, CopyButton } from '@redpanda-data/ui';
+import { computed } from 'mobx';
 
 const AppPageHeader = observer(() => {
     const showRefresh = useShouldShowRefresh();
 
-    return <Box> {/* we need to refactor out #mainLayout > div rule, for now I've added this box as a workaround */}
-        <Flex mb={5} alignItems="center" justifyContent="space-between">
-            <Breadcrumb spacing="8px" separator={<ChevronRightIcon/>}>
-                {!isEmbedded() && uiState.selectedClusterName &&
-                    <BreadcrumbItem>
-                        <BreadcrumbLink as={Link} to="/">
-                            Cluster
-                        </BreadcrumbLink>
-                    </BreadcrumbItem>
-                }
-                {uiState.pageBreadcrumbs.filter((_, i, arr) => {
-                    const isCurrentPage = arr.length - 1 === i;
-                    return !isEmbedded() || isCurrentPage;
-                }).map((entry, i, arr) => {
-                        const isCurrentPage = arr.length - 1 === i;
-                        const currentBreadcrumbProps: BreadcrumbLinkProps = isCurrentPage ? {
-                            as: 'span',
-                            fontWeight: 700,
-                            fontSize: 'xl',
-                        } : {
-                        };
-
-                        return (
-                            <BreadcrumbItem key={entry.linkTo} isCurrentPage={isCurrentPage}>
-                                <BreadcrumbLink
-                                    to={entry.linkTo}
-                                    as={isCurrentPage ? 'span' : Link}
-                                    noOfLines={1}
-                                    {...currentBreadcrumbProps}
-                                    {...(entry.options?.canBeTruncated ? {
-                                        wordBreak: 'break-all',
-                                        whiteSpace: 'break-spaces',
-                                    } : {
-                                        whiteSpace: 'nowrap'
-                                    })}
-                                >
-                                    {entry.title}
-                                </BreadcrumbLink>
-
-                                <Flex ml={2}>
-                                    {isCurrentPage && entry.options?.canBeCopied && <CopyButton content={entry.title} variant="ghost"/>}
-
-                                    {isCurrentPage && showRefresh && (
-                                        <Box minW={250}>
-                                            <DataRefreshButton/>
-                                        </Box>
-                                    )}
-                                </Flex>
-
-                            </BreadcrumbItem>
-                        );
-                    }
+    const breadcrumbItems = computed(() => {
```

Comment:
I don't think a `computed` should be created within an observable component.
Are you sure it won't just always be recreated/recomputed on every render? I think it will...

Instead consider this (outside of a component):

```
const obj = observable({

    get breadcrumbItems() {
          // compute items here
         return items;
    }
});

const AppPageHeader = observer(() => {
       const items =  obj.breadcrumbItems;
       ...
}

```

Reply:
I though creating an computed with `get()` is OK  as part of the observable. I think we use the same concept just class-based elsewhere. There is also an example on Mobx site https://mobx.js.org/computeds-with-args.html#2-close-over-the-arguments

But maybe I misunderstand something. I've tried your solution and it seems the number of renders is identical. I'm happpy to chat about this on Slack.